pkg/workloads/console/controller.go

```diff
@@ -206,6 +226,74 @@ func (r *reconciler) updateStatus(job *batchv1.Job) error {
 	return nil
 }
 
+func calculateStatus(csl *workloadsv1alpha1.Console, job *batchv1.Job, pod *corev1.Pod) workloadsv1alpha1.ConsoleStatus {
+	newStatus := csl.DeepCopy().Status
+
+	// We may have been passed an empty Job struct, if the job no longer exists,
+	// so determine whether it's a real job by checking if it has a name.
+	if job != nil && len(job.Name) != 0 {
+		// We want to give the console session *at least* the time specified in the
+		// timeout, therefore base the expiry time on the job creation time, rather
+		// than the console creation time, to take into account any delays in
+		// reconciling the console object.
+		// TODO: We may actually want to use a base of when the Pod entered the
+		// Running phase, as image pull time could be significant in some cases.
+		jobCreationTime := job.ObjectMeta.CreationTimestamp.Time
+		expiryTime := metav1.NewTime(
+			jobCreationTime.Add(time.Second * time.Duration(csl.Spec.TimeoutSeconds)),
+		)
+		newStatus.ExpiryTime = &expiryTime
+		if pod != nil {
+			newStatus.PodName = pod.ObjectMeta.Name
+		}
+
+		newStatus.Phase = calculatePhase(job, pod)
+	}
+
+	return newStatus
+}
+
+func calculatePhase(job *batchv1.Job, pod *corev1.Pod) workloadsv1alpha1.ConsolePhase {
+	// Currently a job can only have two conditions: Complete and Failed
+	// Both indicate that the console has stopped
+	for _, c := range job.Status.Conditions {
+		if c.Type == batchv1.JobComplete || c.Type == batchv1.JobFailed {
+			return workloadsv1alpha1.ConsoleStopped
```

Comment:
I think in the long term we'll want to set a different phase depending on whether the job completed or failed,  to provide a nicer client-side UX. But for now this should get us going